File: lotteryticket/src/main/java/com/wuhai/lotteryticket/utils/TreeSetNew.java
package com.wuhai.lotteryticket.utils;

import androidx.annotation.NonNull;

import java.util.Iterator;
import java.util.TreeSet;

public class TreeSetNew<E> extends TreeSet<E> {

    @NonNull
    @Override
    public String toString() {
        Iterator<E> it = iterator();
        if (! it.hasNext())
            return "";

        StringBuilder sb = new StringBuilder();
        for (;;) {
            E e = it.next();
            sb.append(e == this ? "(this Collection)" : e);
            if (! it.hasNext())
                return sb.toString();
            sb.append(',');
        }

    }
}

File: lotteryticketkotlin/src/main/java/com/wuhai/demo/lotteryticketkotlin/utils/TreeSetNew.kt
package com.wuhai.demo.lotteryticketkotlin.utils

import java.util.*

class TreeSetNew<E> : TreeSet<E>() {
    override fun toString(): String {
        val it: Iterator<E?> = iterator()
        if (!it.hasNext()) return ""
        val sb = StringBuilder()
        while (true) {
            val e = it.next()
            sb.append(if (e === this) "(this Collection)" else e)
            if (!it.hasNext()) return sb.toString()
            sb.append(',')
        }
    }
}